torch_harmonics/csrc/attention/attention_fwd_cuda.cu

```diff
@@ -585,11 +374,13 @@ static void s2_attention_dipatch(int batch_size,
                                  at::Tensor row_off,
                                  at::Tensor col_idx,
                                  at::Tensor quad_weights,
-                                 at::Tensor yP,
-                                 cudaStream_t stream) {
+                                 at::Tensor yP) {
 
     static_assert(0 == (MAX_LOCAL_ARR_LEN & (MAX_LOCAL_ARR_LEN-1)));
 
+    // get stream
```

Comment:
Don't we want to be able to specify the tream from within PyTorch?

Reply:
that is the way to do it. when using a stream context manager, the current stream will be the one from the context manager, otherwise the default stream for forward or backward, depending on when that routine is called.